rtengine/iptransform.cc

```diff
@@ -1244,6 +1259,21 @@ void ImProcFunctions::transformGeneral(bool highQuality, Imagefloat *original, I
                 pLCPMap->correctDistortion(x_d, y_d, w2, h2);
             }
 
+            x_d /= params->perspective.camera_scale;
+            y_d /= params->perspective.camera_scale;
+            if (params->perspective.camera_defish) {
+                double const focal_source = params->perspective.camera_focal_length * maxRadius * 0.0462;
+                double focal_dst = focal_source;
```

Comment:
Why this copy?

Reply:
The whole geometric transformation code was already twice in this file, once for the actual transformation and once for figuring out which source pixels are needed when showing some zoomed in portion of the image. If I don't add the fisheye code in both places the preview doesn't work when zooming in.

I think the code duplication should be avoided but I didn't look into it so I don't know if it's easy to put it into a single function and call it in both places.

Reply:
Sorry, my thinking was more mundane: `focal_dst` isn't changed in this scope, if I'm not mistaken, so why not use `focal_source` right away?

Reply:
Ah, that was a relic from my early experiments, I fixed it.